kernel/src/main/java/org/kframework/parser/concrete2kore/disambiguation/VariableTypeInferenceFilter.java

```diff
@@ -372,11 +372,20 @@ public String toString() {
 
     public static boolean isFunctionRule(TermCons tc, boolean isAnywhere) {
         if ((tc.production().sort().name().equals("#RuleContent") || tc.production().sort().name().equals("#RuleBody")) && !(tc.get(0) instanceof TermCons && isFunctionRule((TermCons)tc.get(0), isAnywhere))) {
+            if (tc.get(0) instanceof Ambiguity) {
+              return false;
+            }
```

Comment:
Why is it the case that if there is an `Ambiguity` we know this isn't a function rule? Seems to me we could have ambiguities in function rules too.

Reply:
We don't; but when we don't know whether or not it's a function rule, we don't know that it's safe to apply the typing restrictions associated with function rules, so we don't apply them. it's a "degrade gracefully" kind of deal. it seems that the ambiguities in this example are resolved later in the ambiguity pipeline though.

I might be able to implement a case where if all of the ambiguous possibilities are function rules, we also return true. But at worst this is going to fail to catch a type error rather than just crashing, so it seems to me like an improvement.

Reply:
We could also always return true in this case, in which case you would be guaranteed not to miss type errors, but in some cases might be unable to write certain (correct) rules. This seemed like a better solution to me...

Reply:
Let's at least put a TODO here marking that it's a known issue, and (potentially?) add a card to the Parser column of our board mentioning it: https://github.com/kframework/k/projects/2#column-5804675